crates/rattler_installs_packages/src/python_env/venv.rs

```diff
@@ -8,10 +8,26 @@ use crate::python_env::{
     system_python_executable, FindPythonError, ParsePythonInterpreterVersionError,
     PythonInterpreterVersion,
 };
+use std::ffi::OsStr;
+use std::fmt::Debug;
+use std::fs;
+use std::io::ErrorKind;
 use std::path::{Path, PathBuf};
 use std::process::{Command, Output};
 use thiserror::Error;
 
+#[cfg(unix)]
+pub fn copy_file<P: AsRef<Path>, U: AsRef<Path>>(from: P, to: U) -> std::io::Result<()> {
+    std::os::unix::fs::symlink(from, to)?;
+    Ok(())
+}
+
+#[cfg(windows)]
+pub fn copy_file<P: AsRef<Path>, U: AsRef<Path>>(from: P, to: U) -> std::io::Result<()> {
+    fs::copy(from, to)?;
+    Ok(())
+}
```

Comment:
Just wondering if it possible to hardlink this? Not sure about concurrent file/executable access under windows though.

Reply:
Very good question.

Based on my little research, for windows rust uses `CopyFileEx`.  Based on docs, https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-copyfileexw#parameters
this function can receive `dwCopyFlags` and specifically `COPY_FILE_OPEN_SOURCE_FOR_WRITE`  `The file is copied and the original file is opened for write access.` which will tell windows to allow other write access.
This is the call on rust side 
```
c::CopyFileExW(
            pfrom.as_ptr(),
            pto.as_ptr(),
            Some(callback),
            &mut size as *mut _ as *mut _,
            ptr::null_mut(),
            0,
        )
```
( no copy flags passing ) which I understand is that we will not have our file open to other writes.
Also https://source.winehq.org/source/dlls/kernelbase/file.c#0494 based on wine source code I see that wine even don't support this flag which give me confidence that it's concurrent write safe.


On other hand, `CreateHardLink` ( which is used by Rust to create hard links ) https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-createhardlinka
is only supported on the NTFS file system and based on rust docs ( https://doc.rust-lang.org/std/fs/fn.hard_link.html )  it require that these two paths to both be located on the same filesystem.

I think it will be more safe to use `copy` instead of `hardlink` but I would like to ask your opinion

Reply:
Seems like a good idea!